lib/galaxy/webapps/galaxy/api/roles.py

```diff
@@ -65,26 +63,26 @@ class FastAPIRoles:
     role_manager: RoleManager = Depends(get_role_manager)
 
     @router.get('/api/roles')
-    def index(self, trans: SessionRequestContext = Depends(get_trans)) -> RoleListModel:
+    def index(self, trans: ProvidesUserContext = Depends(get_trans)) -> RoleListModel:
```

Comment:
Maybe move the `RoleListModel` definition (line 43) to `galaxy/manager/roles` and import it alongside the other pydantic models?

Reply:
I get why that seems more wholistic and consistent. My counter argument is that I think the distinction is that RoleListModel exists only to adapt a list of roles to the web framework - the managers themselves are more useful and better when they return a list of role models. So it feels like moving that there is pushing "web app concerns" into "app" logic.

Reply:
No, I agree, you're right. In fast, I had similar misgivings about the `TourList` model being part of the tours "schema". I think once we have a good sample of endpoints done, we may go back and refactor for consistency.

Reply:
Finding the right distinction here is messy - I think we're asking the right questions and things are structured better than ever before inside the app - but we will need to work on this and accept a lot of fuzziness in these distinctions.

Thanks fr the detailed review!